src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -83,6 +83,27 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+  public List<Message> getAllMessages(){
+    List<Message> messages = new ArrayList<>();
+
+    Query query = new Query("Message")
+      .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+        String idString = entity.getKey().getName();
+        UUID id = UUID.fromString(idString);
+        String user = (String) entity.getProperty("user");
+        List<Message> userMessages = getMessages(user); //get all messages from specific user
```

Comment:
You may be trying to reuse the `getMessages()` function because the instructions suggested to `encapsulate the redundant code in a function that you can call in both places`. I think that what they meant by that is to create a helper function that contains the redundant (i.e. repetitive) code and call the helper function in both places.